var/spack/repos/builtin/packages/synapsetool/package.py

```diff
@@ -50,6 +50,8 @@ class Synapsetool(CMakePackage):
     depends_on('hdf5~mpi', when='~mpi')
     depends_on('highfive+mpi', when='+mpi')
     depends_on('highfive~mpi', when='~mpi')
+    depends_on('libsonata+mpi', when='+mpi')
```

Comment:
I'm more inclined to have this dependency optional, so we could merge the PR right away even before libsonata is ready.

Reply:
Must admit I'm a bit confused.
If `synapsetool` CI is based on Spack (is it?), how would I be able to pass CI w/o adding `libsonata` as `synapsetool` dependency here?

Reply:
Also, this change merely declares `libsonata` as a dependency; so as long as that one is compilable in Spack world, it won't affect `synapsetool` (from this point of view, the PR _can_ be merged right away).

Or do you mean making this dependency optional in `synapsetool` itself; and guiding corresponding build parameter with a variant here?

Reply:
As is, libsonoata must be available for anyone to build synapsetool, even if we ask for synapsetool 0.1. So that is conditional for synapsetool >= xx (eventually 0.4)
But initially my comment was in the direction that you address in the last comment. If the user doesnt want sonata, let's allow him, and that would be useful now in the beginning, so we could even merge this before the apps are ready, and later change the default to True.

Reply:
The problem for version can be be solved by:

```
    depends_on('libsonata+mpi', when='@0.4:+mpi')
    depends_on('libsonata~mpi', when='@0.4:~mpi')
```

Add variant  `sonata` if you guys want to add this as optional and then : 

```
    depends_on('libsonata+mpi', when='@0.4:+mpi+sonata')
    depends_on('libsonata~mpi', when='@0.4:~mpi+sonata')
```

Reply:
I guess `+sonata` would suffice? (since there is no such configuration as `@:0.4+sonata`, i.e. we can ask `+sonata` only from 0.4+).